uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/my-leave/components/my-leave-calendar_test.js

```diff
@@ -0,0 +1,44 @@
+(function (CRM) {
+  define([
+    'common/angular',
+    'common/angularMocks',
+    'leave-absences/shared/config',
+    'leave-absences/my-leave/app'
+  ], function (angular) {
+    'use strict';
+
+    describe('myLeaveCalendar', function () {
+      var $compile, $log, $rootScope, component, controller;
+
+      beforeEach(module('leave-absences.templates', 'my-leave'));
+      beforeEach(inject(function (_$compile_, _$log_, _$rootScope_) {
+        $compile = _$compile_;
+        $log = _$log_;
+        $rootScope = _$rootScope_;
+
+        spyOn($log, 'debug');
+
+        compileComponent();
+      }));
+
+      it('is initialized', function () {
+        expect($log.debug).toHaveBeenCalled();
+      });
+
+      it('has the legend expanded', function () {
+        expect(controller.legendCollapsed).toBe(false);
+      });
+
+      function compileComponent() {
```

Comment:
I actually do it on purpose, leveraging the hoisting feature of the JS engine. This makes the code more readable, leaving all the functions that you use in the code implementation at the bottom, instead of having them cluttering the code at the beginning